builtin/commit.c

```diff
@@ -1435,6 +1436,10 @@ static int git_commit_config(const char *k, const char *v, void *cb)
 		config_commit_verbose = git_config_bool_or_int(k, v, &is_bool);
 		return 0;
 	}
+	if (!strcmp(k, "commit.allowEmpty")) {
+		config_commit_allow_empty = git_config_bool(k, v);
+		return 0;
+	}
```

Comment:
@dscho this `!strcmp` is not returning true with `commit.allowEmpty`. But works perfectly with `commit.allowempty`.
So, I am changing the config variable from `commit.allowEmpty` to `commit.allowempty`.